### DISCUSSION TITLE
use custom button for one service only

### DISCUSSION THREAD

**Discussion by Colir** [add2cal/add-to-calendar-button#254] [Q&A] [answered]
Hi,
i would like to know if i can use directly the atcb_action() to generate and download an ics file directly, without opening the overlay and list.

The idea is to use  my own custom button, and by js, trigger atcb to directly generate an ics.
It that possible ?

Thank you

**Comment by jekuer** [comment, marked as answer]
Hi @Colir ,

At the moment this is not possible - unfortunately.
However, it will be in the upcoming V2 of the tool. Stay tuned! 😃 .

Best,
Jens

**Comment by just4fun** [comment]
Hi @jekuer , thanks for the information, I also need this feature.

BTW, for the current version, I found a workaround, which can make this feature happen.

What we are doing is, we have the following CSS.

```css
#atcb-bgoverlay {
  display: none !important;
}
```

Meanwhile we have the following JS code after we click our custom button.

```javascript
atcb_action({...})
document.getElementById('atcb-btn-custom-ical')?.click()
```

@jekuer could you share your thoughts whether the code above is reliable for current version?

Thanks.

**Reply by jekuer** [reply]
Yes, this should work (from a high-level perspective).
However, the upcoming built-in functionality will be definitely more stable.

**Reply by just4fun** [reply]
Thank you jekuer.
